src/modules/royalty.ts

```diff
@@ -0,0 +1,203 @@
+import { ERC20__factory, Royalty, Royalty__factory } from "@3rdweb/contracts";
+import { BigNumber } from "ethers";
+import { ModuleType } from "../common";
+import {
+  Currency,
+  CurrencyValue,
+  getCurrencyMetadata,
+  getCurrencyValue,
+} from "../common/currency";
+import { Module } from "../core/module";
+import { SplitRecipient } from "../types/SplitRecipient";
+
+export interface ISplitsModule {
+  /**
+   * Returns a list of all recipients with their
+   * respective split percentages.
+   *
+   */
+  getAllRecipients(): Promise<SplitRecipient[]>;
+
+  /**
+   * Get the split percentage of a recipient.
+   *
+   * @param address - The address of the recipient.
+   */
+  getRecipientSplitPercentage(address: string): Promise<SplitRecipient>;
+
+  /**
+   * Returns the amount of royalty available for a recipient
+   * to withdraw in the native currency.
+   *
+   * @param address - The address of the recipient to check the balance of.
+   */
+  balanceOf(address: string): Promise<BigNumber>;
```

Comment:
not sure if I want an additional `balance` method as well as `balanceOf` (balance would just get *my* balance if I'm connected, but the more I think about it the more I think we should instead get rid of `balance` everywhere else and force users to explicitly pass the address....)